lib/console_web/controllers/v1/channel_controller.ex

```diff
@@ -134,6 +134,111 @@ defmodule ConsoleWeb.V1.ChannelController do
     end
   end
 
+  def create(conn, %{ "name" => name, "type" => "aws", "topic" => topic, "aws_access_key" => pk, "aws_secret_key" => sk, "aws_region" => region }) do
+    current_organization = conn.assigns.current_organization
+
+    channel_params =
+      %{
+        "credentials" => %{
+          "topic" => topic,
+          "aws_access_key" => pk,
+          "aws_secret_key" => sk,
+          "aws_region" => region
+        },
+        "name" => name,
+        "type" => "aws",
+        "organization_id" => current_organization.id
+      }
+
+    with {:ok, %Channel{} = channel} <- Channels.create_channel(current_organization, channel_params) do
+      channel = attach_devices_and_labels(current_organization, channel)
```

Comment:
there wouldn't be any devices or labels attached if I just created the channel, right?